mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2568,6 +2586,11 @@ LogicalResult LLVM::ZeroOp::verify() {
   return success();
 }
 
+/// Fold a zero operation to a dedicated zero attribute.
+OpFoldResult LLVM::ZeroOp::fold(FoldAdaptor) {
+  return mlir::LLVM::ZeroAttr::get(getContext());
```

Comment:
Ah sorry for being unprecise here. The logic that checks if you can build a zero attribute is useful here! If the type of the attribute is an integer for example it is preferable to fold to an integer attribute which then in turn can fold with a follow up ZExt operation.